src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginViewMargin.cs

```diff
@@ -0,0 +1,238 @@
+            ITagAggregator<InheritanceMarginTag> tagAggregator,
+            Document document)
+        {
+            _textViewHost = textViewHost;
+            _textView = textViewHost.TextView;
+            _tagAggregator = tagAggregator;
+            _optionService = document.Project.Solution.Workspace.Services.GetRequiredService<IOptionService>();
+            _languageName = document.Project.Language;
+            _mainCanvas = new Canvas { ClipToBounds = true };
+            _grid = new Grid();
+            _grid.Children.Add(_mainCanvas);
+            _glyphManager = new InheritanceGlyphManager(
+                textViewHost.TextView,
+                threadingContext,
+                streamingFindUsagesPresenter,
+                classificationTypeMap,
+                classificationFormatMap,
+                operationExecutor,
+                editorFormatMap,
+                listener,
+                _mainCanvas);
+            _refreshAllGlyphs = true;
+            _disposed = false;
+
+            _tagAggregator.BatchedTagsChanged += OnTagsChanged;
+            _textView.LayoutChanged += OnLayoutChanged;
+            _textView.ZoomLevelChanged += OnZoomLevelChanged;
+            _optionService.OptionChanged += OnRoslynOptionChanged;
+
+            _mainCanvas.Width = HeightAndWidthOfMargin;
+
+            _grid.LayoutTransform = new ScaleTransform(
+                scaleX: _textView.ZoomLevel / 100,
+                scaleY: _textView.ZoomLevel / 100);
+            _grid.LayoutTransform.Freeze();
+            UpdateMarginVisibility();
+        }
+
+        private void OnZoomLevelChanged(object sender, ZoomLevelChangedEventArgs e)
+        {
+            _grid.LayoutTransform = e.ZoomTransform;
+            _refreshAllGlyphs = true;
+        }
+
+        private void OnLayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
+        {
+            _glyphManager.SetSnapshotAndUpdate(
+                _textView.TextSnapshot,
+                e.NewOrReformattedLines,
+                e.VerticalTranslation ? _textView.TextViewLines : e.TranslatedLines);
+
+            IList<ITextViewLine> lines = _refreshAllGlyphs ? _textView.TextViewLines : e.NewOrReformattedLines;
+            foreach (var line in lines)
+            {
+                _glyphManager.RemoveGlyph(line.Extent);
+                RefreshGlyphsOver(line);
+            }
+
+            _refreshAllGlyphs = false;
+        }
+
+        private void OnRoslynOptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            if (e.Option == FeatureOnOffOptions.ShowInheritanceMargin || e.Option == FeatureOnOffOptions.InheritanceMarginCombinedWithIndicatorMargin)
+            {
+                UpdateMarginVisibility();
+            }
+        }
+
+        private void UpdateMarginVisibility()
+        {
+            var featureEnabled = _optionService.GetOption(FeatureOnOffOptions.ShowInheritanceMargin, _languageName) != false;
```

Comment:
why != false?

Reply:
FeatureOnOffOptions.ShowInheritanceMargin is bool?
Because we used to use the 'null' to do experiment